protocol/0029-FEES-fees.md

```diff
@@ -107,15 +115,25 @@ If the transfer fails:
 1. If we are in continuous trading mode, than trades should be discarded, any orders on the book that would have been hit should remain in place with previous remaining size intact and the incoming order should be rejected (not enough fees error).
 This functionality requires to match orders and create trades without changing the state of the order book or passing trades downstream so that the execution of the transaction can be discarded with no impact on the order book if needed.
 Other than the criteria whether to proceed or discard, this is exactly the same functionality required to implement [price monitoring](./0032-PRIM-price_monitoring.md).
-2. If we are in auction mode, ignore the shortfall (and see more details below).
+1. If we are in auction mode, ignore the shortfall (and see more details below).
+
+The transfer of fees must be completed before performing the normal post-trade calculations (MTM Settlement, position resolution etc...). The transfers have to be identifiable as fee transfers and separate for the different components.
 
-The transfer of fees must be completed before performing the normal post-trade calculations (MTM Settlement, position resolution etc...). The transfers have to be identifiable as fee transfers and separate for the three components.
+Additionally, a `high_volume_market_maker_rebate` may be necessary which will be taken from the `treasury/buyback_fee` components. This will be calculated as:
+1. Determine whether the maker party of the trade (if there is one) qualifies for a `high volume market maker rebate` and, if so, at what rate.
+1. Calculate the fee that this corresponds to as `high_volume_maker_fee = high_volume_factor * trade_value_for_fee_purposes`.
+1. Take this fee from the `treasury_fee` and `buyback_fee` (protocol restrictions on governance changes ensure that `treasury_fee + buyback_fee >= high_volume_maker_fee` is always true) as a proportion of their relative sizes, i.e.:
+   1. `high_volume_maker_fee = high_volume_factor * trade_value_for_fee_purposes`
+   1. `treasury_fee = treasury_fee * (1 - high_volume_maker_fee / (treasury_fee + buyback_fee))`
+   1. `buyback_fee = treasury_fee * (1 - buyback_fee / (treasury_fee + buyback_fee))`
```

Comment:
Shouldn't it be?
```suggestion
   1. `treasury_fee = treasury_fee * (1 - high_volume_maker_fee * treasury_fee / (treasury_fee + buyback_fee))`
   1. `buyback_fee = buyback_fee * (1 - high_volume_maker_fee * buyback_fee / (treasury_fee + buyback_fee))`
```

Reply:
Have changed these up now, not quite what is in this suggested change but *I think* correct now so please double check :D

Reply:
Goood point, looks good now 👍